src/wui/attack_box.cc

```diff
@@ -186,19 +200,19 @@ void AttackBox::update_attack(bool action_on_panel) {
 	more_soldiers_->set_title(std::to_string(max_attackers));
 }
 
-void AttackBox::init() {
-	assert(node_coordinates_);
+void AttackBox::init(const bool fastclick) {
```

Comment:
I think this can be private.
Maybe extract some meaningful methods instead of having one method with 100+ loc?